compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -987,6 +988,26 @@ public void visit(BLangSimpleVariable varNode) {
         result = varNode;
     }
 
+    @Override
+    public void visit(BLangLetExpression letExpression) {
+        SymbolEnv prevEnv = this.env;
+        this.env = letExpression.env;
+        BLangExpression expr = letExpression.expr;
+        BLangBlockStmt blockStmt = ASTBuilderUtil.createBlockStmt(letExpression.pos);
+        for (BLangVariable var : letExpression.letVarDeclarations) {
+            BLangSimpleVariableDef varDef = createVarDef(((BLangSimpleVariable) var).name.getValue(), var.type,
+                    var.expr, var.pos);
+            blockStmt.addStatement(varDef);
+        }
+        BLangSimpleVariableDef tempVarDef = createVarDef("$temp_var_0$", expr.type, expr, expr.pos);
```

Comment:
Shall we rename the temp variable from "$temp_var_0$" to "$temp_var$"(something like this)?

Reply:
Renamed to "$let_var_%d_$" where d is let expression count.